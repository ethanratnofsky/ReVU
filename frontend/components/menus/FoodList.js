import {View, ScrollView, Text, SafeAreaView} from 'react-native';
import menuPageStyles from './MenuStyles';
import FoodItem from './FoodItem';

const FoodList = () => {
    const list = ["Banana Cakes", "Green Beans", "Vegan Chili", "Sloppy Joe"];
    const timeOfDay = "Breakfast";
    return (
        <View>
            <Text style={menuPageStyles.foodTimeHeader}>{timeOfDay}</Text>
            {list.map((foodItem, index) => {
                return (
                    <FoodItem key={index} item={foodItem} />
                )
            })}
        </View>
    );

}

export default FoodList;