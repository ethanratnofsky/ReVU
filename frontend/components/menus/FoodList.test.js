import React from 'react';
const renderer = require('react-test-renderer');
import FoodList from './FoodList';

describe('<FoodList />', () => {
  it('has 5 children', () => {
    const tree = renderer.create(<FoodList />).toJSON();
    expect(tree.children.length).toBe(5);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<FoodList />).toJSON();
    expect(tree).toMatchSnapshot();
  });