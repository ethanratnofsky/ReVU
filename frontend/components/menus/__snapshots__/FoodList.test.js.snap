// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "borderColor": "#CFAE70",
        "borderWidth": 3,
        "color": "#FFFFFF",
        "fontFamily": "LibreCaslonText_700Bold",
        "fontSize": 18,
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 20,
        "padding": 5,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    Breakfast
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#CFAE70",
        "borderWidth": 3,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 5,
        "padding": 5,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1C1C1C",
          "fontFamily": "LibreCaslonText_400Regular",
        }
      }
    >
      Banana Cakes
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#CFAE70",
        "borderWidth": 3,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 5,
        "padding": 5,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1C1C1C",
          "fontFamily": "LibreCaslonText_400Regular",
        }
      }
    >
      Green Beans
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#CFAE70",
        "borderWidth": 3,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 5,
        "padding": 5,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1C1C1C",
          "fontFamily": "LibreCaslonText_400Regular",
        }
      }
    >
      Vegan Chili
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#CFAE70",
        "borderWidth": 3,
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 5,
        "padding": 5,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1C1C1C",
          "fontFamily": "LibreCaslonText_400Regular",
        }
      }
    >
      Sloppy Joe
    </Text>
  </View>
</View>
`;
